wagtail/admin/viewsets/model.py

```diff
@@ -155,19 +166,25 @@ def get_index_view_kwargs(self, **kwargs):
             "search_fields": self.search_fields,
             "search_backend_name": self.search_backend_name,
             "paginate_by": self.list_per_page,
+            "table_class": ViewSetModelTable,
```

Comment:
If we refactor `ViewSetModelTable` into a table class mixin, we can probably add the following property:

```py
@property
def table_class(self):
    if self.sort_order_field:
        return type(f"{self.model.__name__}Table", (OrderableTableMixin, Table), {})
    return Table
```

and then use it here

```suggestion
            "table_class": self.table_class,
```